### ISSUE TITLE
[Suggestion] My JS Code to change the layout without create any Page or Group

### ISSUE THREAD

**Issue by DrBeastMaximus** [matt-kruse/old-layout#16]
My method is create a bookmark with this JS script as URL. Method for every browser that has bookmark bar

`javascript:(()=>{let e=require("DTSGInitialData").token||document.querySelector('[name="fb_dtsg"]').value,t=require("CurrentUserInitialData").USER_ID||document.cookie.match(/c_user=([0-9]+)/)[1];fetch("https://www.facebook.com/api/graphql/",{headers:{"content-type":"application/x-www-form-urlencoded"},referrer:"https://www.facebook.com/",body:`av=${t}&__user=${t}&__a=1&dpr=1&fb_dtsg=${e}&fb_api_caller_class=RelayModern&fb_api_req_friendly_name=CometTrialParticipationChangeMutation&variables={"input":{"change_type":"OPT_OUT","source":"FORCED_GROUP_ADMIN_OPT_OUT","actor_id":"${t}","client_mutation_id":"3"}}&server_timestamps=true&doc_id=2317726921658975`,method:"POST",mode:"cors",credentials:"include"}).then(e=>{console.log("Done"),location.reload()})})();`

and bring it into the bookmark bar for later use. First, access Facebook, then click the bookmark. Facebook will reverse into the old layout for 48hrs. Simply just a click.
Rep if the js code works. But im sure it will ;)

**Comment by Xetaxheb** [comment]
Uncaught SyntaxError: Unexpected token {

**Comment by Xetaxheb** [comment]
(()=>{let e=document.body.getElementsByTagName("script")[6].innerText.match(/\"USER_ID\":\"(.*?)\",\"/m)[1];let t=(()=>{let e=document.body.getElementsByTagName("script");for(let t=0;t<e.length;t++)if(e[t].innerText.includes("fb_dtsg"))return e[t].innerText.match(/\"name\":\"fb_dtsg\",\"value\":\"(.*?)\"}/m)[1]})();fetch("https://www.facebook.com/api/graphql/",{headers:{accept:"*/*","accept-language":"vi-VN,vi;q=0.9,fr-FR;q=0.8,fr;q=0.7,en-US;q=0.6,en;q=0.5","content-type":"application/x-www-form-urlencoded","sec-fetch-dest":"empty","sec-fetch-mode":"cors","sec-fetch-site":"same-origin","viewport-width":"1920"},referrer:"https://www.facebook.com/",referrerPolicy:"no-referrer-when-downgrade",body:`av=${e}&__user=${e}&__a=1&dpr=1&fb_dtsg=${encodeURIComponent(t)}&fb_api_caller_class=RelayModern&fb_api_req_friendly_name=CometTrialParticipationChangeMutation&variables=%7B%22input%22%3A%7B%22change_type%22%3A%22OPT_OUT%22%2C%22source%22%3A%22SETTINGS_MENU%22%2C%22actor_id%22%3A%22${e}%22%2C%22client_mutation_id%22%3A%221%22%7D%7D&server_timestamps=true&doc_id=2317726921658975`,method:"POST",mode:"cors",credentials:"include"}).then(()=>console.log("Done!")).catch(e=>console.error(e))})();

paste in console or use bookmarklet then refresh

**Comment by DrBeastMaximus** [comment]
> (()=>{let e=document.body.getElementsByTagName("script")[6].innerText.match(/"USER_ID":"(._?)","/m)[1];let t=(()=>{let e=document.body.getElementsByTagName("script");for(let t=0;t<e.length;t++)if(e[t].innerText.includes("fb_dtsg"))return e[t].innerText.match(/"name":"fb_dtsg","value":"(._?)"}/m)[1]})();fetch("[https://www.facebook.com/api/graphql/",{headers:{accept:"*/*","accept-language":"vi-VN,vi;q=0.9,fr-FR;q=0.8,fr;q=0.7,en-US;q=0.6,en;q=0.5","content-type":"application/x-www-form-urlencoded","sec-fetch-dest":"empty","sec-fetch-mode":"cors","sec-fetch-site":"same-origin","viewport-width":"1920"},referrer:"https://www.facebook.com/",referrerPolicy:"no-referrer-when-downgrade",body:`av=${e}&__user=${e}&__a=1&dpr=1&fb_dtsg=${encodeURIComponent(t)}&fb_api_caller_class=RelayModern&fb_api_req_friendly_name=CometTrialParticipationChangeMutation&variables=%7B%22input%22%3A%7B%22change_type%22%3A%22OPT_OUT%22%2C%22source%22%3A%22SETTINGS_MENU%22%2C%22actor_id%22%3A%22${e}%22%2C%22client_mutation_id%22%3A%221%22%7D%7D&server_timestamps=true&doc_id=2317726921658975`,method:"POST",mode:"cors",credentials:"include"}).then(()=>console.log("Done!")).catch(e=>console.error(e))})(](https://www.facebook.com/api/graphql/%22,%7Bheaders:%7Baccept:%22*/*%22,%22accept-language%22:%22vi-VN,vi;q=0.9,fr-FR;q=0.8,fr;q=0.7,en-US;q=0.6,en;q=0.5%22,%22content-type%22:%22application/x-www-form-urlencoded%22,%22sec-fetch-dest%22:%22empty%22,%22sec-fetch-mode%22:%22cors%22,%22sec-fetch-site%22:%22same-origin%22,%22viewport-width%22:%221920%22%7D,referrer:%22https://www.facebook.com/%22,referrerPolicy:%22no-referrer-when-downgrade%22,body:%60av=$%7Be%7D&__user=$%7Be%7D&__a=1&dpr=1&fb_dtsg=$%7BencodeURIComponent(t)%7D&fb_api_caller_class=RelayModern&fb_api_req_friendly_name=CometTrialParticipationChangeMutation&variables=%7B%22input%22%3A%7B%22change_type%22%3A%22OPT_OUT%22%2C%22source%22%3A%22SETTINGS_MENU%22%2C%22actor_id%22%3A%22$%7Be%7D%22%2C%22client_mutation_id%22%3A%221%22%7D%7D&server_timestamps=true&doc_id=2317726921658975%60,method:%22POST%22,mode:%22cors%22,credentials:%22include%22%7D).then(()=%3Econsole.log(%22Done!%22)).catch(e=%3Econsole.error(e))%7D)());
> 
> paste in console or use bookmarklet then refresh

yep. You have to open Facebook, and execute the script right on the opened Facebook tab.

**Comment by dmitriy1980** [comment]
Neither suggestion above works for me.
